app/models/user.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  has_one :profile, dependent: :destroy
+
+  # User is post author
+  has_many :posts, foreign_key: :author_id, dependent: :destroy
+
+  # User is post receiver
+  has_many :posts, as: :postable, dependent: :destroy
+
+  has_many :comments, foreign_key: :author_id, dependent: :destroy
+
+  has_many :likes, foreign_key: :liker_id, dependent: :destroy
+
+  has_many :active_friendships, class_name: 'Friendship', foreign_key: :user_id
+
+  has_many :passive_friendships, class_name: 'Friendship', foreign_key: :friend_id
+
+  has_many :added_friends, through: :active_friendships,
+                           source: :friend
+
+  has_many :adding_friends, through: :passive_friendships, source: :user
+
+  has_many :active_requests, class_name: 'FriendRequest', foreign_key: :requester_id
+
+  has_many :passive_requests, class_name: 'FriendRequest', foreign_key: :requestee_id
+
+  has_many :requestees, through: :active_requests
+
+  has_many :requesters, through: :passive_requests
```

Comment:
Assuming we don't merge FriendRequest with Friendship, these lines were necessary because a User can either be a person who sends a friend request (in this case he is 'requester') and a person who receives a friend request (in this case he is a 'requestee').  We distinguished between "active_request" and "passive_request".  

Removing these aliases from the association will mean eager_loading the :friend_requests to User (i.e User.eager_load(:friend_requests)). So this means calling User.first.requestees will show all the FriendRequests that the User.first belongs to (all requests where the User.first is the requester OR the requestee). So when we call on User.first.requestees this will not result to what we want - it will show all the requestees inside the FriendRequest that User.first is involved in. Thus, it will even show User.first as part of that result (as a requestee). So our solution was, in order to show only those users that User.first has sent requests to, is to distinguish between active friend requests (friend requests that User.first sent) and passive_request (friend requests that User.first received). 

I will be sending a pull request to Development from another branch. This branch will contain changes mainly for reducing database queries (using eager_loads and preload for polymorphic associations). Also in that branch we eliminated the distinction between passive_requests and active_requests (just plain friend_requests association for User now). Please note that for these changes we haven't updated the specs yet. And also we haven't incorporated the merging of the User and Profile, and also the Friendship and FriendRequest. Will examine those after the initial comments above.